gulpfile.js

```diff
@@ -762,5 +776,146 @@ function listReleaseTasks(appDirectory) {
+            done();
+        });
+    }
+    return distTasks;
+}
+function cordova_apps() {
+    const appsTasks = [];
+    const platforms = getPlatforms();
+    if (platforms.indexOf('android') !== -1) {
+        appsTasks.push(cordova_build);
+    } else {
+        appsTasks.push(function cordova_dist_none(done) {
+            done();
+        });
+    }
+    return appsTasks;
+}
+
+
+function cordova_clean() {
+    return del(['./cordova/www/**', './cordova/resources/**', './cordova/plugins/**', './cordova/platforms/**'], { force: true });
+}
+function cordova_copy_www() {
+    return gulp.src(`${DIST_DIR}**`, { base: DIST_DIR })
+        .pipe(gulp.src('./changelog.html', { passthrougth: true }))
+        .pipe(gulp.dest(`${CORDOVA_DIR}www/`));
+}
+function cordova_locales_www(cb) {
+    fs.renameSync(`${CORDOVA_DIR}www/_locales`, `${CORDOVA_DIR}www/i18n`);
+    gulp.src(`${CORDOVA_DIR}www/js/localization.js`)
+        .pipe(replace('/_locales', './i18n'))
+        .pipe(gulp.dest(`${CORDOVA_DIR}www/js`));
+    cb();
+}
+function cordova_resources() {
+    return gulp.src('assets/android/**')
+        .pipe(gulp.dest(`${CORDOVA_DIR}resources/android/`));
+}
+function cordova_include_www() {
+    return gulp.src(`${CORDOVA_DIR}www/main.html`)
+        .pipe(replace('<!-- CORDOVA_INCLUDE', ''))
+        .pipe(replace('CORDOVA_INCLUDE_END -->', ''))
+        .pipe(gulp.dest(`${CORDOVA_DIR}www`));
+}
+function cordova_packagejson() {
+    return gulp.src(`${CORDOVA_DIR}package.json`)
+        .pipe(jeditor({
+            'name': pkg.name,
+            'description': pkg.description,
+            'version': pkg.version,
+            'author': pkg.author,
+            'license': pkg.license,
+        }))
+        .pipe(gulp.dest(CORDOVA_DIR));
+}
+function cordova_configxml() {
+    return gulp.src([`${CORDOVA_DIR}config.xml`])
+        .pipe(xmlTransformer([
+            { path: '//xmlns:name', text: pkg.productName },
+            { path: '//xmlns:description', text: pkg.description },
+            { path: '//xmlns:author', text: pkg.author },
+        ], 'http://www.w3.org/ns/widgets'))
+        .pipe(xmlTransformer([
+            { path: '.', attr: { 'version': pkg.version } },
+        ]))
+        .pipe(gulp.dest(CORDOVA_DIR));
+}
+function cordova_exec(chdir, commandname, callback) {
+    let command;
+    switch (commandname) {
+        case 'npm':
+            command = 'npm install --production';
+            break;
+        case 'cordova_platform':
+            command = 'cordova platform add android';
+            break;
+        case 'cordova_run':
+            command = 'cordova run';
+            break;
+        case 'cordova_build':
+            command = 'cordova build --release --buildConfig';
+            break;
+        default:
+            command = '';
+            break;
+    }
+    process.chdir(chdir);
+    child_process.exec(command, function (err, stdout, stderr) {
```

Comment:
Do you think it would be possible to use something like https://www.npmjs.com/package/gulp-cordova-plugin to avoid having to call into the operating system to run cordova?

Reply:
I think so. There are some packages for gulp helping cordova build. I will see if i can improve that.

Reply:
All in all, I just used the cordova-lib and yarn. So there is no more child_process.exec

Reply:
Nice!